src/redux/books/books.js

```diff
@@ -1,33 +1,35 @@
+import axios from 'axios';
+
 const addBook = (payload) => ({ type: 'ADD', payload });
 const removeBook = (id) => ({ type: 'REMOVE', id });
+const getBooks = (payload) => ({ type: 'GET', payload });
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/DwJiw8QxAUA6baUHq2qU/books';
+
+const initialState = [];
 
-const initialState = [
-  {
-    id: '1',
-    title: 'The Horizon',
-    author: 'Sam Felice',
-  },
-  {
-    id: '2',
-    title: 'The Horizon Part 2',
-    author: 'Sam Felice',
-  },
-];
+export const fetchBooks = () => async (dispatch) => {
+  const res = await axios.get(url);
+  const books = [];
+  Object.keys(res.data).map((key) => books.push({ item_id: key, ...res.data[key][0] }));
+  dispatch(getBooks(books));
+};
 
 export const postBook = (payload) => (dispatch) => {
-  dispatch(addBook(payload));
+  axios.post(url, payload).then(() => dispatch(addBook(payload)));
 };
 
 export const deleteBook = (id) => (dispatch) => {
-  dispatch(removeBook(id));
+  axios.delete(`${url}/${id}`).then(() => dispatch(removeBook(id)));
 };
 
 const changeBooks = (state = initialState, action) => {
   switch (action.type) {
     case 'ADD':
       return [...state, action.payload];
+    case 'GET':
+      return action.payload;
     case 'REMOVE':
-      return state.filter((book) => book.id !== action.id);
+      return state.filter((book) => book.item_id !== action.id);
     default:
       return state;
   }
```

Comment:
- [ ] For this one,  `createAsyncThunk` is required to be used in the project for creating thunk action creators when fetching books, adding, and deleting books to and from the API. Kindly use the Redux createAsyncThunk hook that's used for interacting with external API in Redux for all API actions 👍